Docs: Update section on inspecting tables

wypoon opened the pull request:
Using the SparkSessionCatalog to query metadata tables is possible in Spark 3.2.

wypoon commented:
@kbendick @RussellSpitzer @jackye1995 mind reviewing this?

RussellSpitzer commented on docs/versioned/spark/spark-queries.md:
```diff
@@ -168,7 +168,7 @@ To inspect a table's history, snapshots, and other metadata, Iceberg supports me
 
 {{< hint info >}}
-As of Spark 3.0, the format of the table name for inspection (`catalog.database.table.metadata`) doesn't work with Spark's default catalog (`spark_catalog`). If you've replaced the default catalog, you may want to use `DataFrameReader` API to inspect the table. 
+In Spark 3.0 and 3.1, if you have replaced Spark's default catalog (`spark_catalog`) with Iceberg's `SparkSessionCatalog`, you cannot use it to query metadata tables, as the form of the metadata table name (`catalog.database.table.metadata`) is not accepted by Spark. This is fixed in Spark 3.2 by [SPARK-34209](https://issues.apache.org/jira/browse/SPARK-34209). For Spark 3.0 and 3.1, you can conf ...
```
I would just make the wording a bit more specific here. There are a lot of pronouns which I think is a bit difficult to reason about. I also don't think we need the Jira link since that would mostly help dev's and note users, but feel free to keep it if you think it adds value.

In Spark 3.0 and 3.1 the Spark Session catalog (`spark_catalog`) does not support table names with multipart identifiers such as `catalog.database.table.metadata`. To work around the lack of support of multipart identifiers, configure a non-session catalog using the Iceberg SparkCatalog class or use the Spark `DataFrameReader` API. Spark 3.2 and onwards supports multipart identifiers in the Spark Session catalog.

wypoon replied:
Thanks for the review and suggestion.

wypoon commented:
I have updated the text. @RussellSpitzer I tweaked your suggested text slightly, and dropped the reference to the SPARK JIRA.

kbendick commented on docs/versioned/spark/spark-queries.md:
```diff
@@ -168,7 +168,7 @@ To inspect a table's history, snapshots, and other metadata, Iceberg supports me
 
 {{< hint info >}}
-As of Spark 3.0, the format of the table name for inspection (`catalog.database.table.metadata`) doesn't work with Spark's default catalog (`spark_catalog`). If you've replaced the default catalog, you may want to use `DataFrameReader` API to inspect the table. 
+In Spark 3.0 and 3.1, the Spark session catalog (`spark_catalog`) does not support table names with multipart identifiers such as `catalog.database.table.metadata`. To work around this, for querying metadata tables, configure a different catalog that uses the Iceberg `SparkCatalog` class, or use the Spark `DataFrameReader` API. From Spark 3.2 onwards, the session catalog supports table names with ...
```
Nit: Consider `Prior to Spark 3.2` instead of `Spark 3.0 and Spark 3.1`.

I don't think this applies at all to Spark 2.4, but I can still imagine getting questions about it.

wypoon replied:
Spark 2.4 does not support querying tables using SQL, so this does not apply to 2.4 at all. That is why I wrote "In Spark 3.0 and 3.1", since querying using Spark SQL is only supported in Spark 3 onward.
One can argue both ways about "Prior to Spark 3.2": The reader knows that SQL is not supported for Spark 2, so implicitly "prior to Spark 3.2" means 3.0 and 3.1. Or "prior to Spark 3.2" implies that this section applies to 2.4, which would be misleading.

wypoon replied:
@kbendick I updated the text.
I added a line about Spark 2.4, pointing the reader to the section on "Inspecting with DataFrames".
I also used "prior to 3.2".

wypoon commented:
@RussellSpitzer @kbendick mind reviewing this again? I have updated the text and rebased on master (after Sam Redai's reorg of docs). I hope it can be merged now.

wypoon commented:
@RussellSpitzer @kbendick mind reviewing this again? I have updated the text and rebased on master (after Sam Redai's reorg of docs). I hope it can be merged now.

szehon-ho commented on docs/spark/spark-queries.md:
```diff
@@ -168,7 +168,9 @@ To inspect a table's history, snapshots, and other metadata, Iceberg supports me
-As of Spark 3.0, the format of the table name for inspection (`catalog.database.table.metadata`) doesn't work with Spark's default catalog (`spark_catalog`). If you've replaced the default catalog, you may want to use `DataFrameReader` API to inspect the table. 
+For Spark 2.4, use the `DataFrameReader` API to [inspect tables](#inspecting-with-dataframes).
+
+For Spark 3, prior to 3.2, the Spark session catalog (`spark_catalog`) does not support table names with multipart identifiers such as `catalog.database.table.metadata`. To work around this, for querying metadata tables, configure a different catalog that uses the Iceberg `SparkCatalog` class, or use the Spark `DataFrameReader` API. From Spark 3.2 onwards, the session catalog supports table names ...
```
Do we need 'spark_catalog' quote here?   It seems a bit confusing, as we are talking about Spark Session Catalog.  It was probably from before, when mentioning the default catalog.  If not needed I feel we can remove it.

szehon-ho replied:
Also I feel like it is still a bit wordy, can we just say

As a workaround, configure the catalog `org.apache.iceberg.spark.SparkCatalog` or use the DataFrameReader API.

It's more consistent with the previous sentence.  I was thinking we dont need 'for querying metadata tables', as I think that's what this section is about.   And I feel, no need to mention From Spark 3.2 onwards, as it's implied the issue is gone after 3.2.

wypoon replied:
@szehon-ho thanks for the review.
I think that to be correct, one should say "configure a catalog that uses `org.apache.iceberg.spark.SparkCatalog`". The catalog is `prod` or `iceberg_catalog` or whatever one names it. Otherwise, I have adopted your suggestion.

Note: If you don't feel that the link for "session catalog" (in my update) is necessary, I'm fine with removing it.

szehon-ho commented on docs/spark/spark-queries.md:
```diff
@@ -168,7 +168,9 @@ To inspect a table's history, snapshots, and other metadata, Iceberg supports me
-As of Spark 3.0, the format of the table name for inspection (`catalog.database.table.metadata`) doesn't work with Spark's default catalog (`spark_catalog`). If you've replaced the default catalog, you may want to use `DataFrameReader` API to inspect the table. 
+For Spark 2.4, use the `DataFrameReader` API to [inspect tables](#inspecting-with-dataframes).
+
+For Spark 3, prior to 3.2, the Spark [session catalog](../spark-configuration#replacing-the-session-catalog) does not support table names with multipart identifiers such as `catalog.database.table.metadata`. As a workaround, configure a catalog that uses `org.apache.iceberg.spark.SparkCatalog`, or use the Spark `DataFrameReader` API.
```
The second sentence is still a bit long and not too clear (configure a catalog that uses the Iceberg 'SparkCatalog' class) , can we say

```
As a workaround, configure the Iceberg `SparkCatalog` class...
```

I am thinking that especially with the link, it should be enough for them to know how to do it

wypoon replied:
I have to disagree. I think the second sentence is both clear and correct.
Does "configure a catalog using `org.apache.iceberg.spark.SparkCatalog`" sound better to your ears?

wypoon replied:
Ok, I changed it to simply "configure an `org.apache.iceberg.spark.SparkCatalog`".

About the link, earlier in the same file, there is a link "(../spark-configuration#using-catalogs)"; I followed the same syntax. The links don't work in github, but they should be correct on the Apache Iceberg website.

szehon-ho replied:
Yea that works (can we change it to 'a' as org.apache... does not start with a vowel?).  I see what you mean, but I guess confusion to me was , 'a catalog using the class org.apache.iceberg.spark.SparkCatalog', seemed to indicate another catalog that uses this internally.

OK I hope so about the link.

szehon-ho commented:
Thanks, the link is fine with me if it works!  I added one more small rewording, then I think the change will be ok

szehon-ho reviewed (approved):
Looks good to me, @samredai if you want to have a sanity check?

samredai reviewed (approved):
LGTM! I like the conciseness reached after the reviews, the final wording is great.

szehon-ho commented:
Merged, thanks @wypoon for the doc, and all the reviewers @kbendick @RussellSpitzer @samredai

wypoon commented:
Thanks @szehon-ho!
@samredai is it possible to get this update on the website? Iceberg 0.13 supports Spark 3.2, so this information is relevant and helpful to users of the current release.

samredai commented:
@wypoon if you open this change as a hotfix directly against the 0.13.1 branch in the apache/iceberg-docs repo, that will get auto-deployed once it's merged.
